File: progetto/python/blog/models/Token.py
from blog.models.Database import db

class Token(db.Model):
    __tablename__ = "tokens"

    token_id = db.Column(db.BigInteger, primary_key=True)

    token = db.Column(db.String(255), nullable=False)

    expire_date = db.Column(db.DateTime, nullable=False)

    revoked_date = db.Column(db.DateTime, nullable=True)

    owner = db.Column(db.BigInteger, db.ForeignKey("users.user_id"), nullable=False)

File: progetto/java/src/main/java/com/projects/blog/models/Token.java
package com.projects.blog.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "tokens")
public class Token {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long token_id;

    @Column(nullable = false)
    private String token;

    @Column(nullable = false)
    private LocalDateTime expire_date;

    private LocalDateTime revoked_at;

    @ManyToOne(optional = false, fetch = FetchType.EAGER)
    private User owner;

    public Boolean isRevoked(){
        return revoked_at != null;
    }
}
